src/modules/navigator/navigator_main.cpp

```diff
@@ -631,12 +637,46 @@ Navigator::task_main()
 		case vehicle_status_s::NAVIGATION_STATE_AUTO_RTL:
 			_pos_sp_triplet_published_invalid_once = false;
 
-			// if RTL is set to use a mission landing and mission has a planned landing, then use MISSION
-			if (mission_landing_required() && on_mission_landing()) {
-				navigation_mode_new = &_mission;
+			switch (rtl_type()) {
+			case 1:
```

Comment:
Which should define these somewhere.

Reply:
yes.  also "rtl_type" is a better param name than "rtl_land_type" ;-)